assets/js/modules/analytics/util/index.js

```diff
@@ -111,7 +112,9 @@ export function extractAnalyticsDataForPieChart( reports, options = {} ) {
 			rowData.push( tooltipCallback( row, rowData ) );
 		}
 
-		dataMap.push( rowData );
+		if ( ! ( hideEmpty && users < 1 ) ) {
+			dataMap.push( rowData );
+		}
```

Comment:
We should be able to wrap the whole block around the `rowData` variable in this clause, right now the `rowData` variable logic starting in line 110 may be unnecessarily run.